Dockerfile

```diff
@@ -30,26 +33,23 @@ RUN zypper addrepo https://download.opensuse.org/repositories/isv:SUSE:Edge:Edge
 # 5. Network configurator
 RUN zypper install -y \
     xorriso squashfs  \
-    libguestfs kernel-default e2fsprogs parted gptfdisk btrfsprogs guestfs-tools lvm2 \
+    libguestfs kernel-default-base e2fsprogs parted gptfdisk btrfsprogs guestfs-tools lvm2 \
     podman \
     createrepo_c \
     helm \
-    nm-configurator
+    nm-configurator && \
+    zypper clean -a
 
 # TODO: Install nmc via zypper once an RPM package is available
-RUN curl -o nmc-aarch64 -L https://github.com/suse-edge/nm-configurator/releases/download/v0.2.0/nmc-linux-aarch64 && \
-    chmod +x nmc-aarch64 && \
-    curl -o nmc-x86_64 -L https://github.com/suse-edge/nm-configurator/releases/download/v0.2.0/nmc-linux-x86_64 && \
-    chmod +x nmc-x86_64 && \
-    cp nmc-$(uname -m) /usr/local/bin/nmc
-
-RUN curl -o hauler-amd64.tar -L https://github.com/rancherfederal/hauler/releases/download/v0.4.3/hauler_0.4.3_linux_amd64.tar.gz && \
-    tar -xf hauler-amd64.tar && \
-    mv hauler hauler-x86_64 && \
-    curl -o hauler-arm64.tar -L https://github.com/rancherfederal/hauler/releases/download/v0.4.3/hauler_0.4.3_linux_arm64.tar.gz && \
-    tar -xf hauler-arm64.tar && \
-    mv hauler hauler-aarch64 && \
-    cp hauler-$(uname -m) /usr/local/bin/hauler
+RUN curl -o /usr/local/bin/nmc -L https://github.com/suse-edge/nm-configurator/releases/download/${NMCVERSION}/nmc-linux-$(uname -m) && \
+    chmod +x /usr/local/bin/nmc
+
+# Hauler doesn't provide "uname -m" releases (x86_64/aarch64), but golang style ones (amd64/arm64) so we can use podman version instead
+RUN mkdir -p /tmp/hauler && \
+    curl -o /tmp/hauler/hauler.tar -L https://github.com/rancherfederal/hauler/releases/download/v${HAULERVERSION}/hauler_${HAULERVERSION}_$(podman version -f '{{.Client.OsArch}}' | sed 's!/!_!g').tar.gz && \
```

Comment:
This will very soon be replaced with package from OBS (similar to nmc). We could merge this but I just want to bring up that this installs only one of the architectures and not both (which is currently intended).

Reply:
I'd rather not merge this since we know it's going to change. No need to bite off added risk of breaking things that already work for a change that we know isn't going to last long.